lms/djangoapps/grades/new/course_grade_factory.py

```diff
@@ -118,10 +126,21 @@ def iter(
 
     def _iter_grade_result(self, user, course_data, force_update):
         try:
-            method = CourseGradeFactory().update if force_update else CourseGradeFactory().create
-            course_grade = method(
-                user, course_data.course, course_data.collected_structure, course_key=course_data.course_key,
-            )
+            if force_update:
```

Comment:
nitpick: if you wanted to cut down on the lines of code, you could have a kwargs dict that holds all of the common kwargs, add the conditional `method = ...` line back, then do:
```python
if force_update:
    kwargs['force_update_subsections'] = True
course_grade = method(**kwargs)
```